src/components/dashboard/main-content/course-enrollments/CourseEnrollments.jsx

```diff
@@ -43,9 +47,19 @@ const CourseEnrollments = ({ children }) => {
   } = useContext(UserSubsidyContext);
 
   const [assignments, setAssignments] = useState([]);
+  const [showCancelledAssignmentsAlert, setShowCancelledAssignmentsAlert] = useState(true);
+  const [showExpiredAssignmentsAlert, setShowExpiredAssignmentsAlert] = useState(true);
+
   useEffect(() => {
     const data = redeemableLearnerCreditPolicies?.flatMap(item => item?.learnerContentAssignments || []);
-    setAssignments(data);
+    const assignmentsData = sortAssignmentsByAssignmentStatus(data);
+    setAssignments(assignmentsData);
+
+    const hasCancelledAssignments = assignmentsData?.some(assignment => assignment.state === 'cancelled');
+    const hasExpiredAssignments = assignmentsData?.some(assignment => isAssignmentExpired(assignment));
+
+    setShowCancelledAssignmentsAlert(hasCancelledAssignments);
+    setShowExpiredAssignmentsAlert(hasExpiredAssignments);
   }, [redeemableLearnerCreditPolicies]);
 
   const allocatedAssignments = assignments?.filter((assignment) => assignment?.state === ASSIGNMENT_TYPES.allocated
```

Comment:
Is it possible for you to change the `allocatedAssignments` to `allocatedAndAcceptedAssignments` or some reasonable short name?